### PR TITLE
Feature / Reusable Cryptographer - pZuswEFC

### PR DESCRIPTION
**Pull request by Zizaco**

### CODE REVIEW

File: lib/utils/cryptographer.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  }
+}
+
+/**
+ * In order to defend agains dictionary and rainbow table attacks.
+ * @see https://en.wikipedia.org/wiki/Salt_(cryptography)
+ * @param {string} input
+ * @return {string} Salted input
+ */
+const _applySalt = (input) => {
```
Comment (pinguet62) [inline lib/utils/cryptographer.js]: Why not `function applySalt (input) {` ?

File: test/utils/cryptographer.spec.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    let hashed = await Cryptographer.hash(input)
+    let decrypted
+    try {
+      decrypted = await Cryptographer.decrypt(hashed)
+    } catch (err) {
+      decrypted = err
+    }
+
+    expect(hashed).to.not.be.equal(input)
+    expect(decrypted).to.not.be.equal(hashed)
```
Comment (pinguet62) [inline test/utils/cryptographer.spec.js]: Where does the `decrypted` value come from?
The returned value from `try` or the error from `catch`?

Comment (Zizaco) [reply test/utils/cryptographer.spec.js]: It doesn't matter. The idea of this test is to make sure that, if for some reason in the future `decrypt` returns something, It should not be the original string of the hash, in other words: The goal is to break If someone changes the `hash` to something reversable, this function will catch it. That was the goal.

File: lib/utils/cryptographer.js (hunk 1 above)
Comment (Zizaco) [reply lib/utils/cryptographer.js]: Sure, why not

File: test/utils/cryptographer.spec.js (hunk 2 above)
Comment (Zizaco) [reply test/utils/cryptographer.spec.js]: Added `// This should not work. That's is the point of the hash` to the testcase

**Pull request merged**